### PR TITLE
🚀 mark: zero

### PR DESCRIPTION
**Pull request by jrans**
### Label Sync MVP
* sync labels from one github repo to another
* fully tested
* hosted at https://label-sync.herokuapp.com
  
Fixes: #3 
Fixes: #5 
Fixes: #6 
Fixes: #12
Fixes: #13
Fixes: #17 
Fixes: #19

### CODE REVIEW

**Comment by nelsonic** [comment]
looking good so far! 👀 👍  😉

File: lib/index.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+server.connection({ port: process.env.PORT });
+
+server.state('token', { isSecure: false });
+
+server.route(routes);
+
+server.register(
+  [{ register: hapiAuth, options: opts },
+    { register: Inert, options: {} }],
```
Comment (nelsonic) [inline lib/index.js]: using `inert` to render _static_ views will mean we cannot display _error_ messages (_or other dynamic content_) in views ...

**Comment by nelsonic** [comment]
@jrans + @jackcarlisle wow this is a beast! 😮 
I will review this evening ... :first_quarter_moon_with_face:

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,8 @@
+root = true
```
Comment (nelsonic) [inline .editorconfig]: does anyone _not_ have these settings by default in their editor? and does having a linter ( `goodparts` ) not enforce them in the project? (_just curious ..._ 😉 )

File: .eslintignore [hunk 3]
```diff
@@ -0,0 +1 @@
+coverage
```
Comment (nelsonic) [inline .eslintignore]: does `goodparts` not _automatically_ ignore `/coverage`  for linting purposes ?

File: README.md [hunk 4]
```diff
@@ -1,29 +1,68 @@
...
-
-## What?
-A quick and easy way to add an existing list of standardised labels to a GitHub
-repo.
+Our main criteria is that it ***MUST*** be quicker than manually adding labels and will offer a [hosted version](https://label-sync.herokuapp.com/) of the app if you don't want to configure/run the app yourself.
 
 ## Who?
-#### Who should use this?
+
 Any developer who has created multiple repositories and values their time ⏰
```
Comment (nelsonic) [inline README.md]: not "_just_" developers... 😉

File: package.json [hunk 5]
```diff
@@ -26,7 +29,17 @@
...
     "istanbul": "^0.4.5",
+    "nock": "^8.1.0",
     "pre-commit": "^1.1.3",
     "tape": "^4.6.2"
+  },
+  "dependencies": {
+    "env2": "^2.1.1",
+    "handlebars": "^4.0.5",
+    "hapi": "^15.1.1",
+    "hapi-auth-github": "^2.0.0",
```
Comment (nelsonic) [inline package.json]: we _might_ want to add some error handling/view via: https://github.com/dwyl/hapi-error ...?

File: lib/copy.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  var userAgent = decoded.agent;
+  var githubToken = decoded.token;
+  var gh = ghReq(githubToken, userAgent);
+  var labelsSet = 0;
+
+  gh({
+    method: 'GET',
+    path: labelsPath(repoData.sourceOwner, repoData.sourceRepo)
+  }, function (err, labels) {
+    hoek.assert(!err, err);
```
Comment (nelsonic) [inline lib/copy.js]: what _happens_ when there _is_ an error...? are we _handling_ it or displaying it to the client? e.g: https://github.com/dwyl/hapi-error ?

File: lib/validate.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+var check = require('./jwt.js');
+
+function goLogin (_, reply) {
+  return reply.redirect('./login');
+}
+
+module.exports = function (callback) {
```
Comment (nelsonic) [inline lib/validate.js]: this could use a `JSDOC` or other comment clarifying what it's _validating_ ...

File: test/fake_reply.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+module.exports = function () {
```
Comment (nelsonic) [inline test/fake_reply.js]: unclear what this `fake_reply` is _for_ ... a `JSDoc` would be _really helpful_ here...

**Comment by nelsonic** [comment]
@jrans / @jackcarlisle the code in this PR is _really good_. My observations are:
+ **Error handling**. We could simply add https://github.com/dwyl/hapi-error and a simple `error_template` and let the plugin display a _user-friendly_ error if one occurs. 😞  >> 😄 
+ **Linting**. Many of the `.js` files don't have `JSDocs` (_or other comments to help beginners understand the code..._) Search the PR for `/**` to see what I mean ... 😕 
+ **Handlebars** is good, but we _could_ try ***Riot*** _this simple app is meant to be a **learning experience** for our `#next` project_ https://github.com/dwyl/hapi-riot 😉 

I'm happy to _merge_ this because it _works_, is tested and is _slick_. 😍 
Please create issues for each of these points so we can address them. thanks! 👍

**Comment by jackcarlisle** [comment]
@nelsonic good stuff, will add issues now! I've been looking through that riot course today and also hapi-riot. It looks like a very similar implementation to handlebars and the syntax doesn't look too difficult to grasp. I think our knowledge of React will help us with the riot concepts as well!

**Comment by jrans** [comment]
@nelsonic thanks! I guess the main aim was to produce the MVP with as few dependencies as possible but yes be very nice to add hapi-error and hapi-riot to a mark 1. Will document any utils that stick around in a separate PR.

Think it's best to have it in now working. Let F&C peoples know about it and request all the features they want through issues. 

Can implement the new features whilst adding the new technologies.

**Comment by jrans** [comment]
@nelsonic added issues that will be fixed by this PR

**Comment by nelsonic** [comment]
@jrans agreed. just wanted to make my thoughts/feedback clear. 👍 
@jackcarlisle can you please create the issues as per the points above. 

merging. so we can get _feedback_! ❤️  ✅  🚀

**Pull request merged**